cmd/argocd/commands/root.go

```diff
@@ -27,6 +29,52 @@ func initConfig() {
 	cli.SetLogLevel(cmdutil.LogLevel)
 }
 
+func NewDefaultArgoCDCommand() *cobra.Command {
+	return NewDefaultArgoCDCommandWithArgs(ArgoCDCLIOptions{
+		PluginHandler: NewDefaultPluginHandler([]string{"argocd"}),
+		Arguments:     os.Args,
+	})
+}
+
+func NewDefaultArgoCDCommandWithArgs(o ArgoCDCLIOptions) *cobra.Command {
+	cmd := NewCommand()
+
+	if o.PluginHandler == nil {
+		return cmd
+	}
```

Comment:
It seems that it never reaches the `return cmd` line as the `PluginHandler` is never `nil` at this point because of: https://github.com/argoproj/argo-cd/pull/20074/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R29

Reply:
It is the intended behaviour. The Find method will search whether the command you are trying to execute is a normal argocd command or not. If it is, the Find method will return the cmd normally.

Reply:
Sorry but your answer seems unrelated to what I pointed out.
Can you please clarify how the code will ever reach line 43?

Reply:
The code is not reaching ilne 43, because plugin handler is never nil. However, the actual cmd will return from the last line. Having said that, even if we remove the nil check for plugin handler, we will still be able to execute the arogcd command and plugin.

Reply:
> The code is not reaching ilne 43, because plugin handler is never nil.

@nitishfy If this is dead code, please remove it from your PR. This is all I am pointing out in this thread.

> we will still be able to execute the arogcd command and plugin.

I understand that but as I said, this is unrelated to this specific thread. Let's discuss this here: https://github.com/argoproj/argo-cd/pull/20074#discussion_r1791949089